pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -376,6 +471,49 @@ func (b *Base) Objects(rsrc interface{}, rsrclabels map[string]string, observed,
 		}
 	}
 
+	var upgradeJob *batchv1.Job
+	for _, item := range reconciler.ObjectsByType(observed, k8s.Type) {
```

Comment:
You can use: k8s/GetItem()

https://github.com/cdapio/cdap-operator/blob/fe748c2ac36b4de7b784a8ad8360fb9c6d469de2/vendor/sigs.k8s.io/controller-reconciler/pkg/reconciler/manager/k8s/manager.go#L358